package/kernel/linux/modules/crypto.mk

```diff
@@ -93,6 +128,66 @@ endef
 $(eval $(call KernelPackage,crypto-ccm))
 
 
+define KernelPackage/crypto-chacha20poly1305
```

Comment:
This depends on crypto-chacha20 and crypto-poly1305, which KCONFIG are `CRYPTO_CHACHA20` and `CRYPTO_POLY1305`.

Reply:
So, you are saying I should add `CRYPTO_CHACHA20` and `CRYPTO_POLY1305` to `KCONFIG` here?

Reply:
I think we are okay, because `CRYPTO_CHACHA20POLY1305` already selects those automatically?

Reply:
Aren't they *selected* by `CRYPTO_CHACHA20POLY1305` in kernel anyway?

Reply:
> I think we are okay, because `CRYPTO_CHACHA20POLY1305` already selects those automatically?

yes it did, but we need extra files, including `chacha(20)_generic.ko` and `poly1305_generic.ko`

Reply:
Yeah I know it's confusing with all the different options here. Let me know if you see an issue. kmod-wireguard is happy with these on my ramips mt7621 target.